src/main/java/org/pac4j/demo/dw/factory/Pac4jFactory2.java

```diff
@@ -0,0 +1,75 @@
+package org.pac4j.demo.dw.factory;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.pac4j.config.client.PropertiesConfigFactory;
+import org.pac4j.core.client.Clients;
+import org.pac4j.core.config.Config;
+import org.pac4j.dropwizard.Pac4jFactory;
+
+import java.util.Map;
+
+/**
+ * This Pac4j factory relies on client properties {@link #clientsProperties} to define all the clients.
+ */
+public class Pac4jFactory2 extends Pac4jFactory {
+
+    private Map<String, String> clientsProperties;
+
+    public Map<String, String> getClientsProperties() {
+        return clientsProperties;
+    }
+
+    @JsonProperty
+    public void setClientsProperties(Map<String, String> clientsProperties) {
+        this.clientsProperties = clientsProperties;
+    }
+
+    @Override
+    public Config build() {
+        Clients clientsComponent = new Clients();
+        Config config = new Config(clientsComponent);
+
+        if (getCallbackUrl() != null) {
+            clientsComponent.setCallbackUrl(getCallbackUrl());
+        }
+        if (getClientNameParameter() != null) {
+            clientsComponent.setClientNameParameter(getClientNameParameter());
+        }
+        clientsComponent.setCallbackUrlResolver(getCallbackUrlResolver());
+        clientsComponent.setAuthorizationGenerators(getAuthorizationGenerators());
+
+        clientsComponent.setClients(getClients());
+
+        final PropertiesConfigFactory propertiesConfigFactory = new PropertiesConfigFactory(clientsProperties);
```

Comment:
clientsProperties can be null:

Exception in thread "main" java.lang.NullPointerException
	at org.pac4j.config.client.PropertiesConfigFactory.getProperty(PropertiesConfigFactory.java:116)
	at org.pac4j.config.client.PropertiesConfigFactory.tryCreateFacebookClient(PropertiesConfigFactory.java:155)
	at org.pac4j.config.client.PropertiesConfigFactory.build(PropertiesConfigFactory.java:121)
	at org.pac4j.demo.dw.factory.Pac4jFactory2.build(Pac4jFactory2.java:44)
	at org.pac4j.dropwizard.Pac4jBundle.run(Pac4jBundle.java:66)
	at org.pac4j.dropwizard.Pac4jBundle.run(Pac4jBundle.java:34)
	at io.dropwizard.setup.Bootstrap.run(Bootstrap.java:200)
	at io.dropwizard.cli.EnvironmentCommand.run(EnvironmentCommand.java:42)
	at io.dropwizard.cli.ConfiguredCommand.run(ConfiguredCommand.java:85)
	at io.dropwizard.cli.Cli.run(Cli.java:75)
	at io.dropwizard.Application.run(Application.java:79)
	at org.pac4j.demo.dw.Pac4JDemoApplication.main(Pac4JDemoApplication.java:19)

I think PropertiesConfigFactory should initialize the internal Map?